### ISSUE TITLE
Think it's not working properly in 1.12.2 ???

### ISSUE THREAD

**Issue by Zygus42** [CD4017BE/VerticallyStackedDimensions#17]
## What happened and What I did:
I stacked End/Overworld/Nether and made no other adjustments.
Picture of me standing at y=1 on top of the Transition Portal Blocks:
![2019-07-31_18 48 06](https://user-images.githubusercontent.com/20757445/62206316-1cd20a80-b3c4-11e9-8e58-7a98dd24f989.png)

I can break them, jump down and then I fall into the Nether.  But I come out at y=125, break 2 blocks below me and then I'm standing on Barrier Alpha blocks:
![2019-07-31_18 52 13](https://user-images.githubusercontent.com/20757445/62206491-82be9200-b3c4-11e9-8447-a989ac0ba66c.png)

If I go upwards again through the hole in the "sky" above me that I broke through from the Overworld, I end up back where i was in Picture1 above.  If I use Creative mode and break through barrier Alpha and dig all the way down, I get to y=0 and see nothing below me at all:
![2019-07-31_18 57 10](https://user-images.githubusercontent.com/20757445/62206738-0e382300-b3c5-11e9-8f45-5eb0b9fa9160.png)

If I fall down into that, I end up in the top of the End, but the sky above me has very odd Transition Blocks:
![2019-07-31_18 58 25](https://user-images.githubusercontent.com/20757445/62206842-417ab200-b3c5-11e9-8ad8-e5c787bd42bf.png)

When I fall through the End, the bottom of the End has very odd Transition Blocks also:
![2019-07-31_18 59 31](https://user-images.githubusercontent.com/20757445/62206906-640ccb00-b3c5-11e9-932c-ada6c1cb67a3.png)

When I fall through that I end up at the top of the Overworld, which is correct and there's nothing weird.

================================================

So I tried it all again in vanilla with nothing but your mod and similar results.  This time I got 2 layers of Netherrack before the 2 layers of Transition Portal, but otherwise acted the same as I had to break the Portal blocks to go into Nether:
![2019-07-31_19 06 49](https://user-images.githubusercontent.com/20757445/62207388-8eab5380-b3c6-11e9-882e-f3736ebce5ad.png)

Ended up above Barrier Alpha, dug down, jumped into the emptyness and ended up at top of the End, but this time the ceiling was all Netherrack:
![2019-07-31_19 09 41](https://user-images.githubusercontent.com/20757445/62207532-ee096380-b3c6-11e9-9f1f-dacf651e54f4.png)

Couldn't see anything below me, jumped down and died at bottom of the End instead of coming out back in the Overworld.  The End felt very short, like it wasn't as high as it usually was - it was hard to tell as there were no blocks at all that I could see.

Is any of this usual behavior and what can I do fix it ?

## Used versions
Minecraft-1.12.2
VerticalDimensionStack-1.12.2-0.1.4
CD4017BE_lib-1.12.2-6.4.6
forge-1.12.2-14.23.5.2838-installer-win

Log (with mods):
[latest.log](https://github.com/CD4017BE/VerticallyStackedDimensions/files/3451402/latest.log)

Log (without mods):
[latest.log](https://github.com/CD4017BE/VerticallyStackedDimensions/files/3451403/latest.log)

**Comment by CD4017BE** [comment]
when placing a `solid dimension` like the nether on top of an `top open dimension` like the end or overworld, make sure to set the upper `transition layer size` in the `top open dimension` (in this case the end) to -2. This will ensure there are only air blocks in the adjacent dimension so the generated portals are all open and don't form floating blocks in the sky.

I don't know where those weird purple glowstone blocks at the bottom of the end come from but they are probably generated by some other mod and likely responsible for the odd portal formations.

The alpha barrier (which is actually a Fiery Bedrock Barrier: WAILA is displaying it wrong) is defined to be generated by the mod config. But that config script line was only meant as an example and should have been commented out. By the way it is set to be only breakable with an obsidian block and a diamond pickaxe or better.
_Not sure, maybe I even left it in intentionally as progression balancing replacement for the nether portal_

**Comment by Zygus42** [comment]
Is that this section ?
	!dimension data for automatic transition layer generation:
	!dim_transitions:<dimId> = [<botMaterial>, <botLayerSize>, <topMaterial>, <topLayerSize>];
	dim_transitions:0 = ["minecraft:stone", 4, "minecraft:air", 0]; !Overworld has stone at floor and air at ceiling
	dim_transitions:-1 = ["minecraft:netherrack", 4, "minecraft:netherrack", 4]; !The Nether has netherrack at both floor and ceiling
	dim_transitions:1 = ["minecraft:air", 0, "minecraft:air", 0]; !The End has air at both floor and ceiling

If so, didn't know you could set the toplayersize to negatives.

**Comment by CD4017BE** [comment]
Yes that's the part and you could also modify these values via the Dimension stack configuration GUI ingame. There the sliders can be moved up to -2 as well.

Just make sure that the size of the matching transition layer in the adjacent dimension is +2 or larger so that the combined size of the transition layers is still positive.

**Comment by Zygus42** [comment]
I can't seem to get rid of this issue at the bottom of the End:
![2019-08-15_09 35 28](https://user-images.githubusercontent.com/20757445/63067201-5d8f5f00-bf40-11e9-971e-0c120379286e.png)

I've tried setting the dim_transitions:1 from positives to 0 to negatives, but the result doesn't change.  

Could it be because I have the Aether dimension (dim  4) stacked underneath the End ?  I haven't set any transitions for dim 4, as it seems to work dropping from dim 1 to dim 4, and the Aether has it's own transition system back to dim 0.

Any suggestions would be very welcome, as it looks very odd.

**Comment by CD4017BE** [comment]
Sorry for the delayed response, I was quite buisy the last week.

What I find especially odd about that, are the purple Glowstone blocks (I guess they are from the Aether mod) that are generated within the portal layer (same height).
The portal layer is generated during the ore generation phase of world gen but assigned a very late ordering priority to make sure other mods don't override it and I assume Aether is not supposed to generate stuff at the bottom of the End anyways. However the regular pattern (looks suspiciously like 16 block grid) means it's likely world generation related.

You are probably more familiar with the features of Aether and can tell me what that block is supposed to do.

**Comment by Zygus42** [comment]
They're Celestial Crystal from Biomes O Plenty - usually only spawn on the underside of the End islands.  Have no real use other than to act as a differently colored Glowstone.
